### PR TITLE
First pass at single-page view for businesses

### PR DESCRIPTION
**Pull request by microfauna**
This PR adds several features:
- Fixes tag search
- Adds a single-page view for businesses

And improves code-quality in two ways:
- Consolidates GraphQL queries into JSON instead of strings, allowing better composability and less breaking behaviour

To run locally:

```
npm i
npm run serve
```
Navigate to localhost:8080/
Search for businesses using tags
Click a business to visit the dedicated page

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,5 +44,21 @@
     "sass": "^1.19.0",
     "sass-loader": "^8.0.0",
     "vue-template-compiler": "^2.6.10"
-  }
+  },
+  "description": "## Running Locally",
```
Comment (arecvlohe) [inline package.json]: Should we have an actual description?

File: package.json [hunk 2]
```diff
@@ -41,5 +44,21 @@
...
+  "main": ".eslintrc.js",
+  "directories": {
+    "test": "tests"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/nativesintech/native-owned-businesses.git"
+  },
+  "keywords": [],
```
Comment (arecvlohe) [inline package.json]: Add keywords or remove them?

File: package.json [hunk 3]
```diff
@@ -41,5 +44,21 @@
...
+  "author": "",
+  "license": "ISC",
```
Comment (arecvlohe) [inline package.json:62]: I am not familiar with ISC license. Is it much different from MIT?

File: public/index.html [hunk 4]
```diff
@@ -5,6 +5,10 @@
     <meta http-equiv="X-UA-Compatible" content="IE=edge">
     <meta name="viewport" content="width=device-width,initial-scale=1.0">
     <link rel="icon" href="<%= BASE_URL %>favicon.ico">
+    <link
+      href="https://api.tiles.mapbox.com/mapbox-gl-js/v0.53.0/mapbox-gl.css"
```
Comment (arecvlohe) [inline public/index.html]: Should we add a comment stating whether this should stay in sync with the version in package.json? With package.json I usually save them as exact versions to make it easier to track what's breaking.

File: src/App.vue [hunk 5]
```diff
@@ -1,5 +1,14 @@
...
     <router-view/>
   </div>
 </template>
+
+<script>
+import Header from '@/components/Header'
+
+export default {
+  components: { Header }
+}
```
Comment (arecvlohe) [inline src/App.vue:13]: I will have to learn this vue stuff, looks simple enough. Name a component to be used inside of the template tags

File: src/components/BusinessPage.vue [hunk 6]
```diff
@@ -1 +0,0 @@
-<!-- Full, expanded page for businesses -->
```
Comment (arecvlohe) [inline src/components/BusinessPage.vue:1]: Is this file needed?

File: src/components/Header.vue [hunk 7]
```diff
@@ -0,0 +1,25 @@
+<template>
+  <router-link
+    to="/"
+    class="flex py-6 bg-white items-center"
+    v-if="!is_embedded"
```
Comment (arecvlohe) [inline src/components/Header.vue:5]: I will need to play around with this but I guess this means if the route is embedded then don't show the header

File: src/queries/index.js [hunk 8]
```diff
@@ -1,72 +1,96 @@
...
-  }
-  territories {
-    territory {
-      id
-      name
-      description_url
+const BUSINESS_FIELDS = [
+  'id', 'name', 'short_description', 'long_description',
+  {
+    locations: { fields: ['id', 'location', 'name', 'postal_address'] }
```
Comment (arecvlohe) [inline src/queries/index.js:7]: I will need to see if this provides autocomplete in the editor. That would be nice

File: src/views/Business.vue [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+      <h2
+        v-if="hasLocations"
+        class="text-xl text-bold pb-2"
+      >
+        Locations
+      </h2>
+      <div
+        class="locations flex flex-row flex overflow-hidden md:flex row border
+        border-gray-600 mb-6 shadow rounded h-64"
+        v-if="hasLocations"
```
Comment (arecvlohe) [inline src/views/Business.vue]: `hasLocations` is duplicated here, is that necessary?

File: src/views/Business.vue [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+              <div class="border-r font-bold px-4 p-2">Address:</div>
+              <div class="p-2 text-gray-800 flex-1">
+                {{location.postal_address ? location.postal_address : 'No Postal Address Provided'}}
+              </div>
+            </div>
+          </div>
+        </div>
+        <MglMap
+          class="h-full border-l border-gray-600 flex-grow flex-1 z-30"
+          :accessToken="accessToken"
```
Comment (arecvlohe) [inline src/views/Business.vue]: I am guessing we need a "real" access token?

File: src/views/Business.vue [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+          :mapStyle.sync="mapStyle"
+          :center="business.locations[0].location.coordinates"
```
Comment (arecvlohe) [inline src/views/Business.vue]: Just to be safe here, can there be a location w/o coordinates? Just checking this doesn't blow up on us

File: src/views/Business.vue [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+          :zoom="9"
+          @load="onMapLoaded"
+        >
+          <MglMarker
+            v-for="location in business.locations"
+            :key="location.id"
+            :coordinates="location.location.coordinates"
+            color="#edf2f7"
```
Comment (arecvlohe) [inline src/views/Business.vue]: Maybe a color worth making into a constant if we need to use it again. If not, no worries

File: src/views/Business.vue [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+import { GET_BUSINESS_BY_ID } from '@/queries'
+import Mapbox from 'mapbox-gl'
+import { MglMap, MglMarker } from 'vue-mapbox'
+
+export default {
+  props: ['id'],
+  components: { BusinessCard, MglMap, MglMarker },
+  data () {
+    return {
+      accessToken: 'pk.eyJ1Ijoibmlja3NhaGxlciIsImEiOiJja2l5MG9zd2UweWdmMnJwM3pmOTBkbnZlIn0.0PlHofmWvqKzjXi_GBHYZg',
```
Comment (arecvlohe) [inline src/views/Business.vue:73]: Should we be hiding this? Make it part of the `.env` setup?

Comment (arecvlohe) [reply src/views/Business.vue:73]: You might have to go through the commits and erase this as well if it's anything sensitive

File: src/views/Business.vue [hunk 14]
```diff
@@ -0,0 +1,121 @@
...
+    }
+  },
+  computed: {
+    hasLocations () {
+      return this.business && this.business.locations.length > 0
+    }
+  },
+  watch: {
+    business () {
+      this.setSelectedLocation(this.business.locations[0])
```
Comment (arecvlohe) [inline src/views/Business.vue:105]: It might not be apparent to another developer that we only are allowing the user to add at most 1 location so that's why we only care about the first one here. Adding a comment might help somewhere in the code

File: src/views/Business.vue [hunk 15]
```diff
@@ -0,0 +1,121 @@
...
+    }
+  },
+  apollo: {
```
Comment (arecvlohe) [inline src/views/Business.vue:108]: This apollo integration is nice

**Review by arecvlohe — DISMISSED** [review]
I added some feedback on the code. Everything looks good, just some comments on maybe some improvements. I am going to look at the integration next and play around with it and see if there is anything that looks off.

File: package.json (hunk 3 above)
Comment (microfauna) [reply package.json:62]: > The ISC license is functionally equivalent to the BSD 2-Clause and MIT licenses, removing some language that is no longer necessary.

Comment (microfauna) [reply package.json:62]: https://choosealicense.com/licenses/isc/

File: src/views/Business.vue (hunk 11 above)
Comment (microfauna) [reply src/views/Business.vue]: There can be, albeit unlikely. I'll add a safety net.

File: src/views/Business.vue (hunk 13 above)
Comment (microfauna) [reply src/views/Business.vue:73]: It's not! It's considered public info - we just need to domain restrict it on the mapbox side.

File: package.json (hunk 1 above)
Comment (microfauna) [reply package.json]: Done!

File: public/index.html (hunk 4 above)
Comment (microfauna) [reply public/index.html]: I wasn't actually sure why this was here - I think it got added at some other stage so I removed it (vue should add it at build time so it still works).

File: src/components/BusinessPage.vue (hunk 6 above)
Comment (microfauna) [reply src/components/BusinessPage.vue:1]: Deleted!

File: src/components/Header.vue (hunk 7 above)
Comment (microfauna) [reply src/components/Header.vue:5]: Yep! It's for an older use-case where AIWPN wanted to embed a list of local-to-their-area / lumbee businesses, I believe.

File: src/queries/index.js (hunk 8 above)
Comment (microfauna) [reply src/queries/index.js:7]: There are a handful of ways to get GraphQL Schema autocomplete but configuring them felt like a process - we can create a ticket

File: src/views/Business.vue (hunk 14 above)
Comment (microfauna) [reply src/views/Business.vue:105]: I think I'll add a followup PR with the business-owner views to clean this up - as I'll be poking at data schemas

**Review by arecvlohe — DISMISSED** [review]

File: public/index.html (hunk 4 above)
Comment (microfauna) [reply public/index.html]: nvm - it's CSS. I moved its import to `main.js` !

**Review by arecvlohe — APPROVED** [review]
